src/protocol/abi/argument_encoding.md

```diff
@@ -0,0 +1,245 @@
+# Argument Encoding
+
+When crafting transaction script data, you must encode the arguments you wish to pass to the script.
+
+## Types
+
+These are the available types that can be encoded in the ABI:
+
+- Unsigned integers:
+  - `u8`, 8 bits.
+  - `u16`, 16 bits.
+  - `u32`, 32 bits.
+  - `u64`, 64 bits.
+- Boolean: `bool`, either `0` or `1` encoded identically to `u8`.
+- B256: `b256`, arbitrary 256-bits value.
+- Address : `address`, a 256-bit (32-byte) address.
+- Fixed size string
+- Array
+- Sum types
```

Comment:
Do we call them sum types elsewhere?  It's a bit of a language nerd term, where devs will just use Enum.

Reply:
porque no los dos?